pkg/scheduler/core/generic_scheduler.go

```diff
@@ -286,21 +282,21 @@ func (g *genericScheduler) Preempt(ctx context.Context, prof *profile.Profile, s
 		klog.Infof("%v potential nodes for preemption, first %v are: %v", len(potentialNodes), len(sample), sample)
 	}
 	var pdbs []*policy.PodDisruptionBudget
-	if g.pdbLister != nil {
-		pdbs, err = g.pdbLister.List(labels.Everything())
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodDisruptionBudget) {
+		pdbs, err = fh.SharedInformerFactory().Policy().V1beta1().PodDisruptionBudgets().Lister().List(labels.Everything())
 		if err != nil {
 			return "", err
 		}
 	}
```

Comment:
lets wrap this in a function that we move later to the postfilter plugin: 

```
func getPodDisruptionBudgets() ([]*policy.PodDisruptionBudget, error) {
  if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodDisruptionBudget) {
    return fh.SharedInformerFactory().Policy().V1beta1().PodDisruptionBudgets().Lister().List(labels.Everything())
}
 return nil, nil
}
```

Reply:
Updated.

Reply:
Thanks, when we move this in the follow up PR, we should get the lister once at postfilter instantiation and avoid constantly checking the feature flag.

Reply:
yup, currently the followup PR already honors featuregate, but using a different style. I will rebase changes from this PR.